sys/cam/scsi/scsi_pass.c

```diff
@@ -1238,7 +1242,9 @@ passcopysglist(struct cam_periph *periph, struct pass_io_req *io_req,
 				xpt_print(periph->path, "%s: copyout of %u "
 					  "bytes from %p to %p failed with "
 					  "error %d\n", __func__, len_to_copy,
-					  kern_ptr, user_ptr, error);
+					  kern_ptr,
+					  (void *)(__cheri_addr uintptr_t)user_ptr,
```

Comment:
Perhaps we should add a capability format specifier to printf, these casts are quite common.

Reply:
The problem is you can't pass a capability via var-args in hybrid.  It's not implemented in the ABI for either MIPS or RISC-V.  @jrtc27 had some ideas such as passing a pointer to the capability instead as the ABI to use implicitly, but that needs some toolchain work.

Reply:
Right, I forgot that was the reason. I don't have a strong opinion on this, I'd say it depends on how long we plan to support hybrid. It wouldn't reduce the diff too much as we would need to change the format string anyway.